### ISSUE TITLE
Duplicate logs are created for providers like fireworks using a mapping from openai error

### ISSUE THREAD

**Issue by Kaboomtastic** [tensorzero/tensorzero#532]
Several providers are using functions like `map_openai_to_*_error` to map openai error types to the provider's error type but this generates a duplicate entry in the log with a message attributing the error to OpenAI first. In this example I provided an invalid api key to a fireworks provider and you can see in the first message that it is an error with a fireworks variant but the error type is OpenAI client

```
{"timestamp":"2024-12-15T19:37:27.152133Z","level":"WARN","message":"Error from OpenAI client: {\"error\":\"unauthorized\"}","spans":[{"function_name":"haiku","variant_name":"None","name":"inference"},{"function_name":"haiku","variant_name":"accounts/fireworks/models/llama-v3p2-3b-instruct","name":"infer"},{"model_name":"accounts/fireworks/models/llama-v3p2-3b-instruct","name":"infer_model_request"}]}
{"timestamp":"2024-12-15T19:37:27.152226Z","level":"WARN","message":"Error from Fireworks client: {\"error\":\"unauthorized\"}","spans":[{"function_name":"haiku","variant_name":"None","name":"inference"},{"function_name":"haiku","variant_name":"accounts/fireworks/models/llama-v3p2-3b-instruct","name":"infer"},{"model_name":"accounts/fireworks/models/llama-v3p2-3b-instruct","name":"infer_model_request"}]}
```

**Comment by virajmehta** [comment]
Good issue! This is caused by the pattern of logging on error construction we adopted a few weeks ago in order to make sure the spans were attached properly. I think the fix might be to just make the client and server errors for inference providers generic, remove all the error type mapping, and be sure to attach spans that include the model name and provider type so the information is present. 

I could imagine this actually reducing the amount of code in T0 by a bit which is a nice side effect. @GabrielBianconi wdyt?

**Comment by virajmehta** [comment]
Yeah! I was gonna do it soon but you're welcome to it. Can review whenever you're ready.

**Comment by virajmehta** [comment]
Closed by #572
